Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -136,8 +157,14 @@ open class ChatChannelListVC: _ViewController,
     override open func setUpLayout() {
         super.setUpLayout()
         view.embed(collectionView)
-        collectionView.addSubview(loadingIndicator)
-        loadingIndicator.pin(anchors: [.centerX, .centerY], to: view)
+        view.embed(emptyView)
+        view.embed(skeletonListView)
+        emptyView.isHidden = true
+        
+        view.addSubview(channelListErrorView)
+        channelListErrorView.topAnchor.pin(equalTo: view.bottomAnchor).isActive = true
+        channelListErrorView.heightAnchor.pin(equalToConstant: channelListErrorViewHeight).isActive = true
+        channelListErrorView.pin(anchors: [.leading, .trailing], to: view)
```

Comment:
I think ideally, we should just also embed the error view, then it is up to the error view, to centre it and add the amount of height needed. The channel list should not know how the error view layout.

Reply:
Totally agree with Nuno here.

Reply:
Done. embedded the layout and show/hide animation related logic in the error view file.